pkg/polaris/azure/options.go

```diff
@@ -83,3 +86,74 @@ func Regions(regions ...string) OptionFunc {
 		return nil
 	}
 }
+
+// ResourceGroup returns an OptionFunc that gives the specified resource group
+// to the options instance. This is currently only needed for archival feature.
+// Support will be added for other features soon.
+func ResourceGroup(resourceGroup *azure.ResourceGroup) OptionFunc {
+	return func(ctx context.Context, opts *options) error {
+		if resourceGroup == nil {
+			return fmt.Errorf("invalid resource group")
+		}
+		if resourceGroup.Name == "" {
+			return fmt.Errorf("invalid name for resource group")
+		}
+		if resourceGroup.Region == "" {
+			return fmt.Errorf("invalid region for resource group")
+		}
+		opts.resourceGroup = resourceGroup
+		return nil
+	}
+}
+
+// FeatureSpecificInfo returns an OptionFunc that gives the specified
+// feature specific information to the options instance.
+// This is currently only needed for archival encryption feature.
+func FeatureSpecificInfo(
+	featureSpecificInfo *azure.FeatureSpecificInfo,
+) OptionFunc {
+	return func(ctx context.Context, opts *options) error {
+		if featureSpecificInfo == nil {
+			return fmt.Errorf("invalid feature specific information")
+		}
+		managedIdentity := featureSpecificInfo.UserAssignedManagedIdentity
+		if managedIdentity != nil {
+			if managedIdentity.Name == "" {
+				return fmt.Errorf("invalid name for managed identity")
+			}
+			if managedIdentity.ResourceGroupName == "" {
+				return fmt.Errorf("invalid resource group name for managed identity")
+			}
+			if managedIdentity.PrincipalID == "" {
+				return fmt.Errorf("invalid principal ID for managed identity")
+			}
+			if managedIdentity.Region == "" {
+				return fmt.Errorf("invalid region for managed identity")
+			}
+		}
+		opts.featureSpecificInfo = featureSpecificInfo
+		return nil
+	}
+}
+
+func verifyOptionsForFeature(opts options, feature core.Feature) error {
+	switch feature {
+	case core.FeatureCloudNativeArchivalEncryption:
+		if opts.featureSpecificInfo == nil ||
+			opts.featureSpecificInfo.UserAssignedManagedIdentity == nil {
+			return fmt.Errorf(
+				"managed identity should be added for archival encryption",
+			)
```

Comment:
Since this a public project we try use the most common style for Go code hence long lines are allowed. :slightly_smiling_face: